internal/devpkg/package.go

```diff
@@ -403,7 +403,7 @@ func (p *Package) Versioned() string {
 }
 
 func (p *Package) IsLegacy() bool {
-	return p.isDevboxPackage() && !p.isVersioned()
+	return p.isDevboxPackage() && !p.isVersioned() && p.lockfile.Source(p.Raw) == ""
```

Comment:
Another possible API could be

`p.lockfile.Get(p.Raw).Source()`  to have lock packages themselves respond to source()

(and just default Source() to be empty string if not set or if package is nil)